### ISSUE TITLE
fixed-step generator

### ISSUE THREAD

**Issue by HaiFengZeng** [sh-lee-prml/PeriodWave#5]
Looking forward the code open source! the paper mentioned modifying the pre-trained CFM generator into fixed-step generator, can u describe more details?

**Comment by sh-lee-prml** [comment]
Hi @HaiFengZeng 

Thanks for your interest. Code will be released after paper notification (on 22. Jan. 2025).

Initially, we pre-trained PeriodWave with continuous time t using the Flow Matching objective (LR: 2e-4), allowing us to sample the waveform from x_t0 using any ODE solvers. 

For faster sampling, we fine-tuned PeriodWave by generating the waveform signal from x_t0 using 2 or 4 fixed step sampling with an Euler Solver. 
We then optimized the generator using adversarial training and Mel reconstruction losses similar to conventional GAN-based neural vocoder. (LR: 2e-5, it is important to decrease the learning rate)

This new training framework for neural vocoders significantly reduces the overall training time.

Thanks

**Comment by HaiFengZeng** [comment]
@sh-lee-prml  Hi, dear friend, have u prepared for releasing the code?

**Comment by sh-lee-prml** [comment]
Hi @HaiFengZeng 

I'm currently tied up with some other tasks. But, I'll do my best to update the code by Feb. 10.

Thank you for your patience.
